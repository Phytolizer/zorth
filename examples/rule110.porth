mem 98 + 1 .

0 while dup 98 < do
  0 while dup 100 < do
    dup mem + , if
      mem 100 + 42 .
    else
      mem 100 + 32 .
    end
    1 mem 100 + 1 1 syscall3

    1 +
  end
  drop

  mem 100 + 10 .
  1 mem 100 + 1 1 syscall3

  // pattern
  mem     , 1 shl
  mem 1 + ,
  bor

  1 while dup 98 < do
    swap 1 shl 7 band
    over mem + 1 + , bor
    2dup 110 swap shr 1 band
    swap mem + swap .
    swap

    1 +
  end
  drop drop

  1 +
end
drop
