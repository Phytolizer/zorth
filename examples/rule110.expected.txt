:i returncode 0
:i argc 0
:b stdin 0

:b stdout 9898
                                                                                                  * 
                                                                                                 ** 
                                                                                                *** 
                                                                                               ** * 
                                                                                              ***** 
                                                                                             **   * 
                                                                                            ***  ** 
                                                                                           ** * *** 
                                                                                          ******* * 
                                                                                         **     *** 
                                                                                        ***    ** * 
                                                                                       ** *   ***** 
                                                                                      *****  **   * 
                                                                                     **   * ***  ** 
                                                                                    ***  **** * *** 
                                                                                   ** * **  ***** * 
                                                                                  ******** **   *** 
                                                                                 **      ****  ** * 
                                                                                ***     **  * ***** 
                                                                               ** *    *** ****   * 
                                                                              *****   ** ***  *  ** 
                                                                             **   *  ***** * ** *** 
                                                                            ***  ** **   ******** * 
                                                                           ** * ******  **      *** 
                                                                          *******    * ***     ** * 
                                                                         **     *   **** *    ***** 
                                                                        ***    **  **  ***   **   * 
                                                                       ** *   *** *** ** *  ***  ** 
                                                                      *****  ** *** ****** ** * *** 
                                                                     **   * ***** ***    ******** * 
                                                                    ***  ****   *** *   **      *** 
                                                                   ** * **  *  ** ***  ***     ** * 
                                                                  ******** ** ***** * ** *    ***** 
                                                                 **      ******   ********   **   * 
                                                                ***     **    *  **      *  ***  ** 
                                                               ** *    ***   ** ***     ** ** * *** 
                                                              *****   ** *  ***** *    ********** * 
                                                             **   *  ***** **   ***   **        *** 
                                                            ***  ** **   ****  ** *  ***       ** * 
                                                           ** * ******  **  * ***** ** *      ***** 
                                                          *******    * *** ****   ******     **   * 
                                                         **     *   **** ***  *  **    *    ***  ** 
                                                        ***    **  **  *** * ** ***   **   ** * *** 
                                                       ** *   *** *** ** ******** *  ***  ******* * 
                                                      *****  ** *** ******      *** ** * **     *** 
                                                     **   * ***** ***    *     ** *********    ** * 
                                                    ***  ****   *** *   **    *****       *   ***** 
                                                   ** * **  *  ** ***  ***   **   *      **  **   * 
                                                  ******** ** ***** * ** *  ***  **     *** ***  ** 
                                                 **      ******   ******** ** * ***    ** *** * *** 
                                                ***     **    *  **      ******** *   ***** ***** * 
                                               ** *    ***   ** ***     **      ***  **   ***   *** 
                                              *****   ** *  ***** *    ***     ** * ***  ** *  ** * 
                                             **   *  ***** **   ***   ** *    ******* * ***** ***** 
                                            ***  ** **   ****  ** *  *****   **     *****   ***   * 
                                           ** * ******  **  * ***** **   *  ***    **   *  ** *  ** 
                                          *******    * *** ****   ****  ** ** *   ***  ** ***** *** 
                                         **     *   **** ***  *  **  * ********  ** * *****   *** * 
                                        ***    **  **  *** * ** *** ****      * *******   *  ** *** 
                                       ** *   *** *** ** ******** ***  *     ****     *  ** ***** * 
                                      *****  ** *** ******      *** * **    **  *    ** *****   *** 
                                     **   * ***** ***    *     ** ******   *** **   *****   *  ** * 
                                    ***  ****   *** *   **    *****    *  ** ****  **   *  ** ***** 
                                   ** * **  *  ** ***  ***   **   *   ** *****  * ***  ** *****   * 
                                  ******** ** ***** * ** *  ***  **  *****   * **** * *****   *  ** 
                                 **      ******   ******** ** * *** **   *  ****  *****   *  ** *** 
                                ***     **    *  **      ******** ****  ** **  * **   *  ** ***** * 
                               ** *    ***   ** ***     **      ***  * ****** *****  ** *****   *** 
                              *****   ** *  ***** *    ***     ** * ****    ***   * *****   *  ** * 
                             **   *  ***** **   ***   ** *    *******  *   ** *  ****   *  ** ***** 
                            ***  ** **   ****  ** *  *****   **     * **  ***** **  *  ** *****   * 
                           ** * ******  **  * ***** **   *  ***    ***** **   **** ** *****   *  ** 
                          *******    * *** ****   ****  ** ** *   **   ****  **  ******   *  ** *** 
                         **     *   **** ***  *  **  * ********  ***  **  * *** **    *  ** ***** * 
                        ***    **  **  *** * ** *** ****      * ** * *** **** ****   ** *****   *** 
                       ** *   *** *** ** ******** ***  *     ********* ***  ***  *  *****   *  ** * 
                      *****  ** *** ******      *** * **    **       *** * ** * ** **   *  ** ***** 
                     **   * ***** ***    *     ** ******   ***      ** **************  ** *****   * 
                    ***  ****   *** *   **    *****    *  ** *     *****            * *****   *  ** 
                   ** * **  *  ** ***  ***   **   *   ** *****    **   *           ****   *  ** *** 
                  ******** ** ***** * ** *  ***  **  *****   *   ***  **          **  *  ** ***** * 
                 **      ******   ******** ** * *** **   *  **  ** * ***         *** ** *****   *** 
                ***     **    *  **      ******** ****  ** *** ******* *        ** ******   *  ** * 
               ** *    ***   ** ***     **      ***  * ***** ***     ***       *****    *  ** ***** 
              *****   ** *  ***** *    ***     ** * ****   *** *    ** *      **   *   ** *****   * 
             **   *  ***** **   ***   ** *    *******  *  ** ***   *****     ***  **  *****   *  ** 
            ***  ** **   ****  ** *  *****   **     * ** ***** *  **   *    ** * *** **   *  ** *** 
           ** * ******  **  * ***** **   *  ***    *******   *** ***  **   ******* ****  ** ***** * 
          *******    * *** ****   ****  ** ** *   **     *  ** *** * ***  **     ***  * *****   *** 
         **     *   **** ***  *  **  * ********  ***    ** ***** ***** * ***    ** * ****   *  ** * 
        ***    **  **  *** * ** *** ****      * ** *   *****   ***   ***** *   *******  *  ** ***** 
       ** *   *** *** ** ******** ***  *     *******  **   *  ** *  **   ***  **     * ** *****   * 
      *****  ** *** ******      *** * **    **     * ***  ** ***** ***  ** * ***    *******   *  ** 
     **   * ***** ***    *     ** ******   ***    **** * *****   *** * ******* *   **     *  ** *** 
    ***  ****   *** *   **    *****    *  ** *   **  *****   *  ** *****     ***  ***    ** ***** * 
   ** * **  *  ** ***  ***   **   *   ** *****  *** **   *  ** *****   *    ** * ** *   *****   *** 
  ******** ** ***** * ** *  ***  **  *****   * ** ****  ** *****   *  **   **********  **   *  ** * 
 **      ******   ******** ** * *** **   *  *******  * *****   *  ** ***  **        * ***  ** ***** 

:b stderr 0

